### DISCUSSION TITLE
Install poetry plugin via pyproject.toml

### DISCUSSION THREAD

**Discussion by BEllis** [python-poetry/poetry#8080] [General]
I'd like users of my plugin to be able to include it in their pyproject.toml instead of having to install it in their global poetry (i.e. using `poetry self add my-plugin`)

From the documentation, I've added this to my client project,

```
[tool.poetry.plugins."poetry.application.plugin"]
my-command-plugin = "poetry_plugin_mine.plugin:MyApplicationPlugin"
```

and I've added the plugin as a dev dependency to the pyproject.toml

I was expecting when I run `poetry` my command would be listed and my plugin would be enabled.

When I run `poetry` it does not list my command and does not appear to be running my plugin at all.

Am I misunderstanding the `tool.poetry.plugins` TOML section? Is what I'm trying to do possible or pragmatic ?

**Comment by BEllis** [comment]
I should mention, when I use `poetry self add my-plugin` my plugin works as I'd expect.

**Comment by ghost** [comment]
@BEllis I guess you should open up an issue/feature request, otherwise this wont get anymore attention :/

**Reply by lfvjimisola** [reply]
We have this issue as well. We can file an issue if @BEllis won't.
